### PR TITLE
Bug 1791789 - Fix for iOS startup crash caused by Glean

### PR DESCRIPTION
**Pull request by travis79**
The tl;dr of the fix: we had recursion wrapped in a loop which I _think_ could have caused two `process()` operations to be active at the same time in certain circumstances. This removes the outer while loop and just uses recursion to spawn a new `process()` operation when the previous upload operation completes.

Note: This also shuffles some iOS tests around to help reduce the intermittent test failures.

### CODE REVIEW

**Comment by travis79** [comment]
The "iOS build and test" task has a single failing test, which is already a known intermittent.

File: glean-core/ios/Glean/Net/HttpPingUploader.swift [hunk 1]
```diff
@@ -150,36 +150,33 @@ public class HttpPingUploader {
...
-                // we don't want to launch multiple uploads at once.
-                // if the upload finishes, we going to launch the next one.
-                return
-            case .wait(let time):
-                sleep(UInt32(time) / 1000)
-                continue
-            case .done:
-                return
             }
+        case .wait(let time):
```
Comment (rosahbruno) [inline glean-core/ios/Glean/Net/HttpPingUploader.swift:172]: The change makes sense and it makes sense that how it was set up before could've caused multiple processes to run at once.

I don't know a lot about this, so this is more of a question for me. Is the only instance when we need to spawn a new process when we hit `.wait`? Just based on the verbiage and the cases I was curious why we wouldn't create a new process on `.done`. Is `.done` just more of a teardown state?

Comment (travis79) [reply glean-core/ios/Glean/Net/HttpPingUploader.swift:172]: `.done` in this case means there are no more pings to send according to glean-core so no need to spawn another `process()`.  The only times this should spawn a new `process()` to handle the next ping is either when the previous ping response is complete or after glean-core has told us to back off (`.wait`).

Comment (rosahbruno) [reply glean-core/ios/Glean/Net/HttpPingUploader.swift:172]: Ok cool, thanks! The changes make sense, but I don't think I have enough context/expertise to give it a proper sign-off.

**Review by badboy — APPROVED** [review]

**Pull request merged**